src/openzaak/components/catalogi/admin/admin_views.py

```diff
@@ -80,6 +81,13 @@ class CatalogusZaakTypeImportSelectView(
     permission_required = "catalogi.add_zaaktype"
     raise_exception = True
 
+    @staticmethod
+    def find_existing(model, value, catalogus):
+        try:
+            return str(model.objects.get(omschrijving=value, catalogus=catalogus).pk)
```

Comment:
IDK about the string. I just removed it and it does not affect my tests.
For 3, I replaced the model with a queryset but that still queries, so I'll make changes again.

2. It should be unique which is kinda true if you included the start and end date since it cannot overlap. But its more of a question of what is the same iotype.

Reply:
Replaces number 3 with a dictionary of dictionaries